legion/engine/core/data/importers/mesh_importers.cpp

```diff
@@ -36,6 +40,141 @@ namespace legion::core::detail
+     * \param accessorType - tinygltf accessorType, Vertex color is expected to come in vec3 or vec4 - will be handled by the function
+     * \param componentType - tinygltf componentType, Vertex color is expected to come in float, unsigned byte or unsigned short - will be handled by the function
+     * \param data - std::vector<color> the destination of the data copy. The vector will be resized to vector.size()+(tinygltf vertex color size)
+     */
+    void handleGltfVertexColor(const tinygltf::Buffer& buffer, const tinygltf::BufferView& bufferView, int accessorType, int componentType, std::vector<math::color>* data)
+    {
+        //colors in glft are in vec3/vec4 float/unsigned byte/unsigned short
+
+        size_t size = data->size();
+        if (accessorType == TINYGLTF_TYPE_VEC3)
+        {
+            // Copy the vertex colors into the vector, keeping in my mind that the vertex color data is only r,g,b
+
+            log::debug("Color accessor type: Vec3");
+            size_t dataIndex = size;
+            if (componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_BYTE)
+            {
+                // Vertex colors in unsigned byte
+                // Currently not supported
+                log::debug("Color component type: u byte");
+                log::warn("Warning: Vert colors for UNSIGNED BYTE not implemented");
+            }
+            else if (componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_SHORT)
+            {
+                // Vertex colors in unsigned short
+                // Currently not supported
+                log::debug("Color component type: u short");
+                log::warn("Warning: Vert colors for UNSIGNED SHORT not implemented");
+            }
+            else if (componentType == TINYGLTF_COMPONENT_TYPE_FLOAT)
+            {
+                // Vertex colors in float
+                log::debug("Color component type: float");
+                data->resize(size + bufferView.byteLength / sizeof(math::vec3));
+                for (int i = 0; i < bufferView.byteLength; i += 3 * sizeof(float))
+                {
+                    float r = *reinterpret_cast<const float*>(&buffer.data.at(i) + bufferView.byteOffset);
+                    float g = *reinterpret_cast<const float*>(&buffer.data.at(i + sizeof(float)) + bufferView.byteOffset);
+                    float b = *reinterpret_cast<const float*>(&buffer.data.at(i + 2 * sizeof(float)) + bufferView.byteOffset);
+                    data->at(dataIndex++) = math::color(r, g, b);
+                }
+            }
+            else
+            {
+                log::warn("Warning: Vert colors were not stored as UNSIGNED BYTE/SHORT or float, skipping");
+            }
+        }
+        else if (accessorType == TINYGLTF_TYPE_VEC4)
+        {
+            // Copy the vertex colors into the vector, keeping in my mind that the vertex color data is r,g,b,a
+
+            log::debug("Color accessor type: Vec4");
```

Comment:
same story, either remove debug log or turn them into trace logs, this is not interesting to a gameplay programmer.